File: 56-merge-intervals/merge-intervals.java
import java.util.*;

class Solution {
    public int[][] merge(int[][] intervals) {
        if (intervals == null || intervals.length == 0) return new int[0][2];
        
        Arrays.sort(intervals, Comparator.comparingInt(a -> a[0]));
        
        List<int[]> merged = new ArrayList<>();
        int[] currentInterval = intervals[0];
        
        for (int i = 1; i < intervals.length; i++) {
            int[] nextInterval = intervals[i];
            
            if (nextInterval[0] <= currentInterval[1]) {
                currentInterval[1] = Math.max(currentInterval[1], nextInterval[1]);
            } else {
                merged.add(currentInterval);
                currentInterval = nextInterval;
            }
        }
        
        merged.add(currentInterval);
        
        int[][] result = new int[merged.size()][2];
        for (int i = 0; i < merged.size(); i++) {
            result[i] = merged.get(i);
        }
        
        return result;
    }
}

File: 56-merge-intervals/merge-intervals.cpp
class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        if (intervals.empty()) return {};
        
        sort(intervals.begin(), intervals.end());
        
        vector<vector<int>> merged;
        
        vector<int> current = intervals[0];
        
        for (const auto& interval : intervals) {
            if (interval[0] <= current[1]) {
                current[1] = max(current[1], interval[1]);
            } else {
                merged.push_back(current);
                current = interval;
            }
        }
        
        merged.push_back(current);
        
        return merged;
    }
};
